src/main/java/unc/Duke.java

```diff
@@ -0,0 +1,198 @@
+                        try {
+                            taskList.add(new Event(oldTask[1], oldTask[2]));
+                        } catch (UncException uncException) {
+                            System.out.println("Error creating saved list.");
+                        }
+                        break;
+                    default:
+
+                }
+            }
+        } catch (FileNotFoundException e) {
+            try {
+                saveFile.getParentFile().mkdir();
+                saveFile.createNewFile();
+            } catch (IOException ioException) {
+                System.out.println(ioException);
+            } catch (SecurityException securityException) {
+                System.out.println(securityException);
+            }
+            System.out.println("Starting new file.");
+        }
+
+
+        String input;
+        Scanner scanner = new Scanner(System.in);
+        while(!Objects.equals(input = scanner.nextLine(), "bye")) {
+            String[] words = input.split("\\s+", 2);
+            try {
+                validInput verb = validInput.valueOf(words[0].toUpperCase());
+                switch (verb) {
+                case LIST:
+                    displayList();
+                    break;
+                case MARK:
+                    mark(Integer.parseInt(words[1]));
+                    break;
+                case UNMARK:
+                    unmark(Integer.parseInt(words[1]));
+                    break;
+                case DELETE:
+                    delete(Integer.parseInt(words[1]));
+                    break;
+                case TODO:
+                    try {
+                        addToDo(words[1]);
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new UncException("You need something to do for unc.task.Todo.");
+                    }
+                    break;
+                case DEADLINE:
+                    addDeadline(words[1]);
+                    break;
+                case EVENT:
+                    addEvent(words[1]);
+                    break;
+                }
+            } catch (UncException uncException) {
+                System.out.println(uncException);
+            } catch (IllegalArgumentException illegalArgumentException) {
+                System.out.println("Invalid command.");
+            } catch (Exception e) {
+                System.out.println(e);
+            }
+
+        }
+
+        System.out.println("Bye");
+        scanner.close();
+        saveFile();
+    }
+
+    private static void displayList() {
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.println(i + 1 + ". " + taskList.get(i));
+        }
+    }
+
+    /*
+    private static void addToList(String input) {
+        System.out.println("added: " + input);
+        taskList.add(new unc.task.Task(input));
+    }
+
+    */
+
+    private static void addToDo(String input) throws UncException {
```

Comment:
This is just nitpicking, but you could consider renaming the method to `addTodo`, especially since the class name is not capitalised in this way as well